rs/http_endpoints/xnet/src/lib.rs

```diff
@@ -208,62 +218,66 @@ fn start_server(
         loop {
             select! {
                 Ok((stream, _peer_addr)) = listener.accept() => {
+                    let metrics = metrics.clone();
+                    metrics.connections_total.inc();
                     let logger = logger.clone();
                     let hyper_service = hyper_service.clone();
-
-                    #[cfg(test)]
-                    {
-                        // TLS is not used in tests.
-                        let _ = tls;
-                        let _ = registry_client;
-
-                        let io = TokioIo::new(stream);
-                        let conn = http.serve_connection_with_upgrades(io, hyper_service);
-                        let wrapped = graceful_shutdown.watch(conn.into_owned());
-                        tokio::spawn(async move {
-                            if let Err(err) = wrapped.await {
+                    let registry_client = registry_client.clone();
+                    let tls = tls.clone();
+                    tokio::spawn(async move {
+                        let http = hyper_util::server::conn::auto::Builder::new(hyper_util::rt::TokioExecutor::new());
```

Comment:
Why create a new server for every request?

Reply:
1. it is not a new server, it is the object that drives the http protocol, this used to be super check
2. also it is not per request, it is per connection

Reply:
oh man what a great question

does hyper_util have a server pool?  would it help?